src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -581,6 +581,21 @@ def test_process_declarations():
     assert isinstance(ptr_sym.datatype, UnsupportedFortranType)
     assert isinstance(ptr_sym.initial_value, IntrinsicCall)
 
+    # Class declaration
```

Comment:
Not your fault at all but this test is huge. Could you break this new part out into a separate test for support for `class` declarations? Please could you also add a test for `class(*)`.

Reply:
Done